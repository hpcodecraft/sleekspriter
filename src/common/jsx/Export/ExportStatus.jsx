// Flux: done, editor: done
var ExportStatus = React.createClass({
  mixins: [FluxMixin],
  getInitialState: function() {
    return {
      timer: null
    }
  },
  render: function() {
    return (
      <div className="bar" ref="bar">{this.props.ui.export.status}</div>
    )
  },
  componentDidUpdate: function() {
    if(this.state.timer === null && this.props.ui.export.status.length > 0) {
      var timer = setTimeout(this.clearStatus, 5000);
      this.setState({timer: timer});
    }
  },
  clearStatus: function() {
    this.setState({timer: null});
    this.getFlux().actions.exportStatus('');
  },
});