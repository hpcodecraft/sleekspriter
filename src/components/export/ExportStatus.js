import React from "react";
import PropTypes from "prop-types";

class ExportStatus extends React.Component {
  state = {
    timer: null,
  };

  render() {
    return <div className="bar">{this.props.status}</div>;
  }

  componentDidUpdate() {
    if (this.state.timer === null && this.props.status.length > 0) {
      var timer = setTimeout(() => this.clearStatus(), 5000);
      this.setState({ timer: timer });
    }
  }

  clearStatus() {
    this.setState({ timer: null });
    this.props.setStatus("");
  }
}

ExportStatus.propTypes = {
  status: PropTypes.string.isRequired,
  setStatus: PropTypes.func.isRequired,
};

export default ExportStatus;
