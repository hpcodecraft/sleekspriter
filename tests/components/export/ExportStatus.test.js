import ExportStatus from "components/export/ExportStatus";

describe("ExportStatus", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      status: "Export finished",
      setStatus: jest.fn()
    };
    wrapper = shallow(<ExportStatus {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("clearStatus", () => {
    it.skip("clears the timer state", () => {
      wrapper.setState({ timer: "some timer" }, () => {
        wrapper.instance().clearStatus(); // TODO: clearStatus uses async setState
        expect(wrapper.state("timer")).toEqual(null);
      });
    });

    it("calls setStatus handler with empty status", () => {
      wrapper.instance().clearStatus();
      expect(props.setStatus).toBeCalledWith("");
    });
  });
});
