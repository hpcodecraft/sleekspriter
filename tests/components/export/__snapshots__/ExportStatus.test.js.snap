// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportStatus should render correctly 1`] = `"<div class=\\"bar\\">Export finished</div>"`;
